src/Client.ts

```diff
@@ -282,14 +282,19 @@ export default class Client {
         return this.publishEnvelope({
           contentTopic: topic,
           message: msgBytes,
+          timestamp,
         })
       })
     )
 
     return this.decodeMessage(msgBytes, topics[0])
   }
 
-  async publishEnvelope(env: messageApi.Envelope): Promise<void> {
+  async publishEnvelope(env: {
+    contentTopic: string
+    message: Uint8Array
+    timestamp?: Date
```

Comment:
I dropped the messageApi.Envelope here so that I don't have to flip back and forth between Date and the string encoding of the ns timestamp.